google/cloud/pubsub/internal/batch_callback.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BATCH_CALLBACK_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BATCH_CALLBACK_H
+
+#include "google/cloud/pubsub/message.h"
+#include "google/cloud/pubsub/version.h"
+#include "google/cloud/completion_queue.h"
+#include "google/cloud/future.h"
+#include "google/cloud/status_or.h"
+#include <google/pubsub/v1/pubsub.pb.h>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * Define the interface to receive message batches from Cloud Pub/Sub via the
+ * Streaming Pull.
+ */
+class BatchCallback {
+ public:
+  virtual ~BatchCallback() = default;
+
+  // Define the struct to store the response from Cloud Pub/Sub.
+  struct StreamingPullResponse {
+    // A batch of messages received.
+    StatusOr<google::pubsub::v1::StreamingPullResponse> response;
+  };
+
+  virtual void operator()(StreamingPullResponse response) = 0;
```

Comment:
Consider giving this a name. Easier to grok `foo->bar(...)` vs. `foo->operator()(...);`

Reply:
Done